src/components/Editor.vue

```diff
@@ -341,6 +342,10 @@ export default {
 				initialSession: this.initialSession,
 			}).catch((e) => {
 				this.hasConnectionIssue = true
+				if (e.response.status === 401) {
+					const redirectUrl = '/index.php/apps/files/?dir=' + encodeURIComponent(this.currentDirectory || '/') + '&openfile=' + this.fileId
```

Comment:
Maybe we can check here if the files app is actually active, as the text app might be rendered in other apps as well through viewer or directly in collectives.